core/storage/user/gae_models.py

```diff
@@ -2215,3 +2215,68 @@ def get_new_id(cls, unused_entity_name):
                 return new_id
 
         raise Exception('New id generator is producing too many collisions.')
+
+
+class UserAuthModel(base_models.BaseModel):
+    """Stores the authentication details for a particular user.
+
+    Instances of this class are keyed by user id.
+    """
+
+    # Authentication detail for sign in using google id (GAE).
+    gae_id = ndb.StringProperty(indexed=True, repeated=True)
```

Comment:
Why repeated=True?

Reply:
To allow more than one gae id (hence email id) be associated with the same user account. Although user settings model would need changes too to store multiple email ids, but I wanted to keep the design here bit general to possibly allow such a case in future.

Reply:
I'm not sure we want to allow that. It seems messy. @BenHenning what do you think?

Reply:
Hmm, I can see cases where we might want to allow changing emails or allowed multiple emails, but we don't need that functionality now (and we'd likely model it post GAE ID world). Given that, I think not having this repeated seems reasonable.

Reply:
Thanks for your insights on this @BenHenning . Yes it seems reasonable to remove the repeated=True part. I will do it and change the tests also accordingly.

Reply:
Change is done now.